python-sdk/src/astro/databases/snowflake.py

```diff
@@ -191,7 +197,8 @@ def set_url_from_file(self, file: File) -> None:
         """
         # the stage URL needs to be the folder where the files are
         # https://docs.snowflake.com/en/sql-reference/sql/create-stage.html#external-stage-parameters-externalstageparams
-        url = file.path[: file.path.rfind("/") + 1]
+        path = file.location.snowflake_stage_path
+        url = path[: path.rfind("/") + 1]
```

Comment:
@utkarsharma2, could we improve this by using stdlib `urlparse`?

Reply:
@tatiana  Yup, done it. please check.